src/views/activity/activity.view.tsx

```diff
@@ -48,19 +49,26 @@ const Activity: FC = () => {
   const onDisplayPending = () => setDisplayAll(false);
 
   const onClaim = (bridge: Bridge) => {
-    setDisabledBridges((current) => [...current, bridge.id]);
     if (bridge.status === "on-hold") {
+      setDisabledBridges([...disabledBridges, bridge.id]);
```

Comment:
Why this change? Isn't this introducing a bug when we have already some bridges disabled?

Reply:
Nope, why would the number of bridges disabled affect the behavior of this setDisabledBridges?

Reply:
No, it's not related to the number of bridges. The problem is that the state we use now is not fresh and therefore if there were already disabled bridges it does not work as expected. Look:


https://user-images.githubusercontent.com/15896177/194038080-81067a17-f53f-4aec-89fe-df0d10897389.mp4

Reply:
Well, in fact the behavior for this should be different. While you are waiting for tx confirmation you shouldn't be able to finalise more bridges. I'll revert it by the moment but I'll create an issue for improving this.